src/platforms/web/runtime/directives/model.js

```diff
@@ -101,6 +108,15 @@ function setSelected (el, binding, vm) {
   }
 }
 
+function hasNoMatchingOption (value, options) {
```

Comment:
Yes, it would be better to use [].some. 
Actually I just did revert changes made in [https://github.com/vuejs/vue/commit/c90b140e80f1bd5d01c733a5bd9fa6e9cb3c2b4d](https://github.com/vuejs/vue/commit/c90b140e80f1bd5d01c733a5bd9fa6e9cb3c2b4d), and didn't think about it. 😅
Will do it later.